adder/local/dag_service.go

```diff
@@ -50,6 +53,7 @@ func (dgs *DAGService) Add(ctx context.Context, node ipld.Node) error {
 			return err
 		}
 		dgs.dests = dests
+		dgs.bph.SetDests(dests)
```

Comment:
dests are used as allocations while pinning.

Also, we can't use BlockPutHelper.dests as allocations for pinnning, since it would not have failed peers.

Reply:
don't want to set a one time thing (BlockPutHelper) in a method that would be called multiple times. 
I actually wanted to move setting `dests` to the constructor.

> Also, chances are the main object does not need to remember dests anymore, since the BlockPutHelper does.

dests are used as allocations while pinning.

Also, we can't use BlockPutHelper.dests as allocations for pinnning, since it would not have failed peers.

Reply:
> dests are used as allocations while pinning.

ok

> don't want to set a one time thing (BlockPutHelper) in a method that would be called multiple times

but you are doing exactly that wth blockputhelper.dests + having an extra method.

Reply:
> I actually wanted to move setting dests to the constructor.

This should make things better